### DISCUSSION TITLE
Using Vue DevTools script could be very dangerous

### DISCUSSION THREAD

**Discussion by alberto-bottarini** [laravel/nova-issues#3198] [Q&A]
- Laravel Version: all
- Nova Version: all
- PHP Version: all

To follow this script https://nova.laravel.com/docs/3.0/customization/frontend.html#vue-devtools could be very dangerous.

`npm install` installs the latest version of `laravel/nova-js` that probably is not the version included in the already-compiled bundle 

I think you should use a fixed version in package.json or commit the package-json.lock (or yarn.lock) file in order to download the very same version used for building bundle.

In my situation I had some bugs when using dev build (because downloaded nova-js was an higher version)

**Comment by ArroWsGM** [comment]
@davidhemphill Any progress with it?

I'm used `addRoute` instead of deprecated `addRoutes` in my development environment, but it does not exist in the production build

I think it's not so complicated to provide `package-lock` for us

**Comment by davidhemphill** [comment]
We use Yarn when building Nova, so I'd recommend using that to get the dependencies from `yarn.lock`. 👍

**Reply by ArroWsGM** [reply]
I notice that but you recommend using npm in the docs, and, as I said, it's not so complicated to provide them both with something like [this package](https://github.com/imsnif/synp).

Or even better if you'd prebuild both and modify the artisan command to publish desired version
